spec/models/miq_expression_spec.rb

```diff
@@ -1829,4 +1829,38 @@
       )
     end
   end
+
+  describe "#field_in_sql?" do
+    it "returns false for model.virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm-archived"
+      expression = {"=" => {"field" => field, "value" => "true"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
+    end
+
+    it "returns false for model.association-virtualfield" do
+      field = "ManageIQ::Providers::InfraManager::Vm.storage-v_used_space_percent_of_total"
+      expression = {">=" => {"field" => field, "value" => "50"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(false)
+    end
+
+    it "returns true for model-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm-vendor"
+      expression = {"=" => {"field" => field, "value" => "redhat"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(true)
+    end
+
+    it "returns true for model.association-field" do
+      field = "ManageIQ::Providers::InfraManager::Vm.guest_applications-vendor"
+      expression = {"CONTAINS" => {"field" => field, "value" => "redhat"}}
+      expect(described_class.new(expression).field_in_sql?(field)).to eq(true)
+    end
+
+    it "returns false if column excluded by preprocessing" do
+      field = "EmsClusterPerformance-cpu_usagemhz_rate_average"
+      expression = {">=" => {"field" => field, "value" => "0"}}
+      obj = described_class.new(expression)
+      obj.preprocess_options = {:vim_performance_daily_adhoc => true}
```

Comment:
It might be better to have an integration test which shows this in action. If you can't do that here, can you update the description of this test to reflect the above? It currently indicates that there's something special about the column, but it was the deliberately setting of this value which caused it to return false.